frontend/src/app/messages/messages.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { environment } from '../../environments/environment';
+
+
+export interface ServerMessage {
+  message: string;
+  type: string;
+}
+
+export class Message {
+  message: string;
+  type: string;
+  dismissible: boolean;
+  /** Timeout is set in seconds, not ms. Can be floating-point. */
+  timeout?: number;
+}
+
+export enum MessageType {
+  Success = 'success',
+  Info = 'info',
+  Warning = 'warning',
+  Danger = 'danger',
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Light = 'light',
+  Dark = 'dark'
+}
+
+export const DEFAULT_TIMEOUT_TIME = 7.5;
+
+@Injectable()
+export class MessagesService {
+  private messageSubject = new Subject<Message>();
+  public message$ = this.messageSubject.asObservable();
```

Comment:
@radoslawroszkowiak what do You think regarding these observable/subject pattern?